processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -919,26 +902,59 @@ fn build_memory_chiplet_responses<E>(main_trace: &MainTrace, row: RowIndex, alph
 where
     E: FieldElement<BaseField = Felt>,
 {
-    let is_read = main_trace.chiplet_selector_3(row);
-    let op_label = get_op_label(ONE, ONE, ZERO, is_read);
+    let element_word = main_trace.chiplet_selector_4(row);
+    let header = {
+        let read_write = main_trace.chiplet_selector_3(row);
+        let op_label = get_memory_op_label(read_write, element_word);
+
+        let ctx = main_trace.chiplet_memory_ctx(row);
+        let clk = main_trace.chiplet_memory_clk(row);
+        let address = {
+            let batch = main_trace.chiplet_memory_batch(row);
+            let idx0 = main_trace.chiplet_memory_idx0(row);
+            let idx1 = main_trace.chiplet_memory_idx1(row);
+
+            batch + idx1.mul_small(2) + idx0
+        };
 
-    let ctx = main_trace.chiplet_memory_ctx(row);
-    let clk = main_trace.chiplet_memory_clk(row);
-    let addr = main_trace.chiplet_memory_addr(row);
-    let value0 = main_trace.chiplet_memory_value_0(row);
-    let value1 = main_trace.chiplet_memory_value_1(row);
-    let value2 = main_trace.chiplet_memory_value_2(row);
-    let value3 = main_trace.chiplet_memory_value_3(row);
+        alphas[0]
+            + alphas[1].mul_base(op_label)
+            + alphas[2].mul_base(ctx)
+            + alphas[3].mul_base(address)
+            + alphas[4].mul_base(clk)
```

Comment:
nit: we do this computation in a few places - maybe makes sense to create a small helper function for it.